### PR TITLE
[HOPSWORKS 3241] add progress bar during stream data ingestion in python

### PR DESCRIPTION
**Pull request by DhananjayMukhedkar**

### CODE REVIEW

File: python/hsfs/engine/python.py [hunk 1]
```diff
@@ -751,9 +752,23 @@ def _write_dataframe_kafka(
...
-            if err is not None:
+            if err is not None and offline_write_options.get("debug_kafka", False):
                 print("Failed to deliver message: %s: %s" % (str(msg), str(err)))
-
+            else:
+                try:
+                    if progress_bar is not None:
+                        progress_bar.update()
+                except Exception as e:
+                    print("Failed tp update progress bar: {}".format(e))
```
Comment (robzor92) [inline python/hsfs/engine/python.py:762]:
```suggestion
                    print("Failed to update progress bar: {}".format(e))
```

**Review by robzor92 — CHANGES_REQUESTED** [review]

File: python/hsfs/engine/python.py [hunk 2]
```diff
@@ -751,9 +752,23 @@ def _write_dataframe_kafka(
         writer = self._get_encoder_func(feature_group._get_encoded_avro_schema())
 
         def acked(err, msg):
-            if err is not None:
+            if err is not None and offline_write_options.get("debug_kafka", False):
                 print("Failed to deliver message: %s: %s" % (str(msg), str(err)))
-
+            else:
+                try:
+                    if progress_bar is not None:
```
Comment (robzor92) [inline python/hsfs/engine/python.py:759]: I think you can skip this try catch block, or are you expecting a specific error?

**Review by robzor92 — CHANGES_REQUESTED** [review]

**Review by moritzmeister — COMMENTED** [review]
@robzor92 do we have to pin the tqdm version? probably not, right, so we are more flexible and it should't be as critical as other libraries.

**Comment by robzor92** [comment]
> 

Not that I'm aware of, but I don't think so, probably some really old version will not be compatible but then alot of other libraries break too probably

File: python/hsfs/engine/python.py [hunk 3]
```diff
@@ -751,9 +752,20 @@ def _write_dataframe_kafka(
         writer = self._get_encoder_func(feature_group._get_encoded_avro_schema())
 
         def acked(err, msg):
-            if err is not None:
+            if err is not None and offline_write_options.get("debug_kafka", False):
                 print("Failed to deliver message: %s: %s" % (str(msg), str(err)))
-
+            else:
+                if progress_bar is not None:
```
Comment (robzor92) [inline python/hsfs/engine/python.py:758]: I don't think we need these `is not None` checks, if it's none then it failed during initialization of the progress bar anyway.

File: python/hsfs/engine/python.py [hunk 4]
```diff
@@ -806,6 +816,8 @@ def acked(err, msg):
 
         # make sure producer blocks and everything is delivered
         producer.flush()
+        if progress_bar is not None:
```
Comment (robzor92) [inline python/hsfs/engine/python.py:819]: I don't think we need these `is not None` checks, if it's none then it failed during initialization of the progress bar anyway.

**Review by robzor92 — CHANGES_REQUESTED** [review]

File: python/hsfs/engine/python.py (hunk 3 above)
Comment (DhananjayMukhedkar) [reply python/hsfs/engine/python.py]: agree, done

File: python/hsfs/engine/python.py (hunk 4 above)
Comment (DhananjayMukhedkar) [reply python/hsfs/engine/python.py]: agree, done

**Review by robzor92 — APPROVED** [review]

**Pull request merged**